### PR TITLE
px.NO_COLOR

### PR DESCRIPTION
**Pull request by nicolaskruchten**
## Code PR

- [x] I have read through the [contributing notes](https://github.com/plotly/plotly.py/blob/master/contributing.md) and understand the structure of the package. In particular, if my PR modifies code of `plotly.graph_objects`, my modifications concern the `codegen` files and not generated files.
- [x] I have added tests (if submitting a new feature or correcting a bug) or
  modified existing tests.
- [x] For a new feature, I have added documentation examples in an existing or
  new tutorial notebook (please see the doc checklist as well).
- [x] I have added a CHANGELOG entry if fixing/changing/adding anything substantial.

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
@emmanuelle the CI failure actually is material, and seems to be https://github.com/plotly/plotly.py/issues/2618

**Comment by nicolaskruchten** [comment]
(but not due to this PR as far as I can tell, so we should not consider it a PR blocker IMO)

**Comment by emmanuelle** [comment]
Looks good to me but we could add documentation examples:
```
import plotly.express as px
df = px.data.experiment(indexed=True)
fig = px.line(df, y=['experiment_1', 'experiment_2'], facet_row='variable', color=px.NO_COLOR)
fig.show()
```
for NO_COLOR, and an example in `subplots.md` for the spacing parameters. Do you want to add them here or to open an issue for this? Otherwise :woman_dancing:

**Comment by nicolaskruchten** [comment]
I'll document the spacing params here. I already have one `NO_COLOR` doc example, do you think we need a second one?

**Comment by emmanuelle** [comment]
Sorry between the beginning and the end of the review I had forgotten about the NO_COLOR example :-). 

Was it intentional to add `px.timeline` in the same PR?

**Comment by nicolaskruchten** [comment]
Yeah, although I probably should be a bit more disciplined. I added `base` to `bar` in this one too as prep for `timeline`

**Comment by nicolaskruchten** [comment]
OK, I've stripped out the bar base and timeline stuff into https://github.com/plotly/plotly.py/pull/2626

**Comment by nicolaskruchten** [comment]
still need a little bit of docs in this PR

File: doc/python/facet-plots.md [hunk 1]
```diff
@@ -115,8 +115,25 @@ fig.for_each_annotation(lambda a: a.update(text=a.text.split("=")[-1]))
...
+df = px.data.gapminder().query("continent == 'Africa'")
+
+fig = px.line(df, x="year", y="lifeExp", facet_col="country", facet_col_wrap=7,
+              facet_row_spacing=0.04, # default is 0.07 when facet_col_wrap is used
+              facet_col_spacing=0.04, # default is 0.03
+              height=600, width=800,
+              title="Life Expectancy in Africa")
+fig.for_each_annotation(lambda a: a.update(text=a.text.split("=")[-1]))
+fig.update_yaxes(showticklabels=True)
+fig.show()
```
Comment (emmanuelle) [inline doc/python/facet-plots.md:136]: great example!

**Comment by emmanuelle** [comment]
:woman_dancing: when CI is green (but 3.7-orca)

**Pull request merged**